lib/walker.ts

```diff
@@ -317,14 +317,37 @@ function stepDetect(
   }
 }
 
-function findCommonJunctionPoint(file: string, realFile: string) {
+/**
+ * Find a common junction point between a symlink and the real file path.
+ *
+ * @param {string} file The file path, including symlink(s).
+ * @param {string} realFile The real path to the file.
+ *
+ * @throws {Error} If no common junction point is found prior to hitting the
+ *                 filesystem root.
+ */
+async function findCommonJunctionPoint(file: string, realFile: string) {
   // find common denominator => where the link changes
-  while (toNormalizedRealPath(path.dirname(file)) === path.dirname(realFile)) {
+  while (true) {
+    const stats = await fs.lstat(file);
+
+    if (stats.isSymbolicLink()) {
+      return { file, realFile };
+    }
+
     file = path.dirname(file);
     realFile = path.dirname(realFile);
-  }
 
-  return { file, realFile };
+    // If the directory is /, break out of the loop and log an error.
+    if (
+      file === path.parse(file).root ||
+      realFile === path.parse(realFile).root
+    ) {
+      throw new Error(
+        'Reached root directory without finding a common junction point',
+      );
```

Comment:
may be worth adding a reference of the two original paths in the error

Reply:
We do not have the original paths anymore at this point.